src/libraries/Microsoft.Extensions.Logging/src/LoggerFactoryScopeProvider.cs

```diff
@@ -48,6 +48,16 @@ void Report(Scope current)
                     }
 
                     callback(activityLogScope, state);
+
+                    if ((_activityTrackingOption & ActivityTrackingOptions.Tags) != 0)
```

Comment:
This logic need to be moved to ActivityLogScope. 
[https://github.com/dotnet/runtime/blob/cbb5b905d67a91ecf293c8807414b7791713b02d/src/libraries/Microsoft.Extensions.Logging/src/LoggerFactoryScopeProvider.cs#L133](https://github.com/dotnet/runtime/blob/cbb5b905d67a91ecf293c8807414b7791713b02d/src/libraries/Microsoft.Extensions.Logging/src/LoggerFactoryScopeProvider.cs#L133)

Reply:
> It makes no sense to cache them like it's done for the ActivityLogScope, as they change from one log line to the other.

I see your point but will be very costly if we request that every time as it allocates. I'll try to think in that and get back to you.

Reply:
I thought about it too before I came to this implementation.

The question is why do you think we benefit of caching it or what do you think its worth to cache? The Tags and Baggage are there anyway. What we do is to enumerate them per call to log.

See e.g. what a formatter does, it just enumerates:
https://github.com/dotnet/runtime/blob/dceccd3a3d33220d93ce3297378bae26744acf2b/src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs#L112

Hence it is not worth to cache the collection somehow.
What in fact does allocate is if the object passed as value is not a string as `ToString()` is called. 
That means if we are able to cache, we only do this once. It's worth to mention that this doesn't matter if the value is already a string. This is the case always for baggage (https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/Activity.cs,80).

Reply:
I agree with you it is not worth it to cache Tags nor Baggage. I was trying to think in a way to optimize nor generating the tags/baggage if it is not changed. But I didn't find straightforward way to do so. What you have now would be ok especially the behavior is opt-in behavior which is clearly will come with a cost.